roxctl/common/environment/auth_config.go

```diff
@@ -0,0 +1,143 @@
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	refreshTokenPath = "/sso/session/tokenrefresh"
+)
+
+type configMethod struct {
+	env Environment
+}
+
+var (
+	_ auth.Method = (*configMethod)(nil)
+)
+
+// ConfigMethod provides an auth.Method for using authentication via local configuration.
+// It will use the configuration store by roxctl central login.
+func ConfigMethod(env Environment) auth.Method {
+	return &configMethod{
+		env: env,
+	}
+}
+
+func (c configMethod) Type() string {
+	return "local configuration"
+}
+
+func (c configMethod) GetCredentials(url string) (credentials.PerRPCCredentials, error) {
+	token, err := c.retrieveToken(url)
+	if err != nil {
+		return nil, errors.Wrap(err, "retrieving token")
+	}
+	return tokenbased.PerRPCCredentials(token), nil
+}
+
+func (c configMethod) retrieveToken(url string) (string, error) {
+	cfgStore, err := c.env.Config()
+	if err != nil {
+		return "", errors.Wrap(err, "retrieving config store")
+	}
+	cfg, err := cfgStore.Read()
+	if err != nil {
+		return "", errors.Wrap(err, "reading configuration")
+	}
+
+	access := cfg.GetCentralConfigs().GetCentralConfig(url).GetAccess()
+	if access == nil || (access.AccessToken == "" && access.RefreshToken == "") {
+		return "", errox.NoCredentials.Newf(
+			`no credentials found for %s, please run "roxctl central login" to obtain credentials`, url)
+	}
+
+	// 1. Check if an access token is given. If an access token is given, and it's not yet expired, then return it.
+	if access.AccessToken != "" && access.ExpiresAt != nil && access.ExpiresAt.After(time.Now().Add(30*time.Second)) {
+		return access.AccessToken, nil
+	}
+
+	// 2. In case an access token is given, but either no expiration is indicated or the token is already expired, and
+	// no refresh token is available, print a warning that the authentication _may_ fail and users have to re-run
+	// roxctl central login in case of authentication issues.
+	if access.AccessToken != "" && access.RefreshToken == "" {
+		c.env.Logger().WarnfLn(`No indication about expiration found or already expired for access token for central %s.
+Still, trying to authenticate with the given token. In case there's any issues, run "roxctl central login" again.`, url)
+		return access.AccessToken, nil
+	}
+
+	// 3. This is the case where an access token is either not set or expired, and a refresh token is available.
+	// We attempt to refresh the access token here.
+	if err := c.refreshAccessToken(url, access); err != nil {
+		return "", errors.Wrap(err, "refreshing access token")
+	}
+
+	// 4. Write the config with the updated data to the store.
+	if err := cfgStore.Write(cfg); err != nil {
+		return "", errors.Wrap(err, "writing configuration to store after refreshing the token")
+	}
+
+	return access.AccessToken, nil
+}
+
+func (c configMethod) refreshAccessToken(url string, accessConfig *config.CentralAccessConfig) error {
+	client, err := c.env.HTTPClient(time.Minute)
```

Comment:
I believe we should pass a context not time

Reply:
That's probably something for a follow-up PR, but I think we shouldn't pass a context here at all, but rather within `HTTPClient.DoReqAndVerifyStatusCode/NewReq`. Since the client creation could be done ahead of time (typically we do this in `Construct` of command), it could be that by the time we execute the client, the context with the timeout is already expired.

I'll create a follow-up for that one.